validator/tests/test_report.py

```diff
@@ -34,5 +35,22 @@ def test_str(self):
         report.add_raised_error(self.ERROR_1)
         report.add_raised_error(self.ERROR_2)
 
-        assert str(report) == "3 checks run\n1 checks failed\n1 checks suppressed\n2 warnings\n"\
-            "bad failed for data 2: msg 2\nwrong import\nright import\n"
+        assert report.summary() ==\
+            "3 checks run\n1 checks failed\n1 checks suppressed\n2 warnings\n"
+
+    @mock.patch("delphi_validator.report.logger")
+    def test_log(self, mock_logger):
+        """Test that the logs contain all failures and warnings."""
+        report = ValidationReport(set([("good", "data 1")]))
+        report.increment_total_checks()
+        report.increment_total_checks()
+        report.increment_total_checks()
+        report.add_raised_warning(ImportWarning("wrong import"))
+        report.add_raised_warning(ImportWarning("right import"))
+        report.add_raised_error(self.ERROR_1)
+        report.add_raised_error(self.ERROR_2)
+
+        report.log()
+        mock_logger.critical.assert_called_once_with(
```

Comment:
I don't feel too strongly, but generally don't feel its necessary to test logging statements (since it's mostly just duplication of the actual code and doesn't really verify state)

Reply:
I feel that the logging here is important to test since it is part of the core functionality of the validator.